index.js

```diff
@@ -44,17 +44,14 @@ module.exports = {
         var ui         = deployment.ui;
         var config     = deployment.config[this.name] = deployment.config[this.name] || {};
 
-        return validateConfig(ui, config)
-          .then(function() {
-            ui.write(blue('|    '));
-            ui.writeLine(blue('- config ok'));
-          });
+        return this._resolvePipelineData(config, context)
+          .then(validateConfig.bind(this, ui, config));
       },
 
       didBuild: function(context) {
         var deployment = context.deployment;
         var ui         = deployment.ui;
-        var config     = deployment.config[this.name] || {};
+        var config     = deployment.config[this.name] = deployment.config[this.name] || {};
```

Comment:
Same here.....My plan is to target that sort of stuff now, once the `activate` is done.....Now that we have 3 plugins doing a similar thing I have a clearer picture about what makes sense to pull out.  So I'd like to do that in a separate PR.  Just starting looking at it now.